src/assignments/tech-jobs-mvc.rst

```diff
@@ -3,4 +3,509 @@
+console application.
+
+This template will be used to display search results, in addition to rendering
+the form. This will give the nice user experience of easily searching multiple
+times in a row.
+
+Wrap Up the Code Review
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once you understand the controllers and views that are already in place, you’re
+ready to begin your work.
+
+In IntelliJ, select *View > Tool Windows > TODO* to pop open a small pane at
+the bottom of the window. This list is populated by any code comments that
+start with ``TODO``. You’ll see your tasks listed, and clicking on any one will
+open the relevant file.
+
+.. figure:: figures/techJobsTodos.png
+   :alt: TechJobs MVC TODO list.
+
+As you work on your tasks, refer to the
+`demo app <https://java-web-dev-techjobs-mvc.cfapps.io/>`__.
+
+2) Complete the List Views
+---------------------------
+
+Open the ``list-jobs.html`` template. Currently, the page just establishes the
+navigation bar and page title. You need to add code that will present relevant
+job information.
+
+Display List of Jobs
+^^^^^^^^^^^^^^^^^^^^^
+
+This page needs to show full job listings---ID, employer, location, etc.
+
+#. In ``list-jobs.html``, create a loop to display each job passed in from the
+   controller. The job data should be presented in a *table*. Be sure to check
+   in ``ListController`` to find the variable names available to the template.
+#. Adding the CSS class ``"job-listing"`` to a table provides some nice
+   styling, courtesy of Carly’s work!
+#. Construct one table, putting each job in a separate row with the fields on
+   separate lines. Alternatively, you can put each job in its own table, with
+   one job field per row.
+
+   .. figure:: figures/list-jobsTableOptions.png
+      :alt: Image of the different table options.
+      :scale: 70%
+
+#. Be sure to test your code by running the program and clicking links from
+   different categories on the ``/list`` page.
+
+   a. Clicking the *Web - Back End* link yields 6 jobs.
+   b. *iOS* yields 4 jobs.
+   c. *New York* yields 1 job.
+   d. Etc.
+
+Add ``View All`` Link
+^^^^^^^^^^^^^^^^^^^^^^
+
+Open the ``list.html`` template. This file builds the table to display all of
+the links for the different ``Job`` fields. However, the link for the ``All``
+column is missing.
+
+.. figure:: figures/listTableOptions.png
+   :alt: Image of the ``/list`` table.
+
+You can fix this several different ways, but the options include:
+
+#. Modify ``tableChoices`` in ``ListController`` to include another key/value
+   pair. Check the method that renders the template to help identify the name
+   to use for the key.
+#. Modify ``list.html`` to fill in the empty table cell with the necessary
+   link. Check ``ListController`` to help identify the data to pass in for the
+   query parameters.
+
+   .. admonition:: Warning
+
+      This is an either/or situation. You only need to implement ONE option, not
+      both.
```

Comment:
Do you mean that they need to pick between steps 1 or 2? You should add that specifically to avoid confusion.

Reply:
I fixed the phrasing to make this point more obvious.